deluge/conftest.py

```diff
@@ -31,6 +33,29 @@ def listen_port(request):
     return DEFAULT_LISTEN_PORT
 
 
+@pytest.fixture
+def mock_callback(timeout=0.5):
+    """Returns a `Mock` object which can be registered as a callback to test against.
+
+    If callback was not called within `timeout` seconds, it will raise a TimeoutError.
+    The returned Mock instance will have a `deferred` attribute which will complete when the callback has been called.
+    """
+
+    def reset():
+        if mock.called:
+            org_reset_mock()
+        deferred = Deferred()
+        deferred.addTimeout(timeout, reactor)
+        mock.side_effect = lambda *args, **kw: deferred.callback((args, kw))
+        mock.deferred = deferred
+
+    mock = unittest.mock.Mock()
+    org_reset_mock = mock.reset_mock
```

Comment:
I might just spell out the whole `orig` here, not saving much space, and it seems more clear.

Reply:
Done